arm-iothub/2016-02-03/swagger/iothub.json

```diff
@@ -0,0 +1,1936 @@
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/subscriptionId"
+        }, {
+          "name": "resourceGroupName",
+          "in": "path",
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/api-version"
+        }],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/IotHubSkuDescriptionListResult"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/IotHubs/{resourceName}/eventHubEndpoints/{eventHubEndpointName}/ConsumerGroups": {
+      "get": {
+        "tags": ["GET"],
+        "summary": "Get EventHub Consumer Groups for an IotHub",
+        "description": "Get EventHub Consumer Groups for an IotHub",
+        "operationId": "IotHubResource_ListEventHubConsumerGroups",
+        "consumes": [],
+        "produces": ["application/json", "text/json"],
+        "parameters": [{
+          "name": "resourceName",
+          "in": "path",
+          "description": "Name of the resource.",
+          "required": true,
+          "type": "string"
+        }, {
+          "name": "eventHubEndpointName",
+          "in": "path",
+          "description": "Name of the event hub endpoint.",
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/subscriptionId"
+        }, {
+          "name": "resourceGroupName",
+          "in": "path",
+          "required": true,
+          "type": "string"
+        }, {
+          "$ref": "#/parameters/api-version"
+        }],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/StringListResult"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false,
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/IotHubs/{resourceName}/eventHubEndpoints/{eventHubEndpointName}/ConsumerGroups/{groupName}": {
+      "get": {
+        "tags": ["GET"],
+        "summary": "Get EventHub Consumer Group for an IotHub",
+        "description": "Get EventHub Consumer Group for an IotHub",
+        "operationId": "IotHubResource_GetEventHubConsumerGroup",
```

Comment:
Well I dont think there is a right answer for this. it really depends on how you want to standardize the format. Having ResourceType_OperationName is as consistent as Having OperationGroup_OperationName. Infact I would prefer something like ResourceType_OperationGroup_OperationName. Is that supported?